jlm/hls/backend/rvsdg2rhls/mem-conv.cpp

```diff
@@ -28,9 +28,10 @@ jlm::hls::route_response(rvsdg::Region * target, jlm::rvsdg::output * response)
   else
   {
     auto parent_response = route_response(target->node()->region(), response);
-    auto ln = dynamic_cast<jlm::hls::loop_node *>(target->node());
-    JLM_ASSERT(ln);
-    auto input = jlm::rvsdg::structural_input::create(ln, parent_response, parent_response->Type());
+    auto lambda = dynamic_cast<jlm::hls::loop_node *>(target->node());
+    JLM_ASSERT(lambda);
```

Comment:
You could use the `AssertedCast<>()` here, then you could throw out the `JLM_ASSERT()`

Reply:
Since this code was not supposed to be touched, I've not made the suggested change.